client/constants/messages.js

```diff
@@ -45,12 +45,24 @@ const messages = {
   FETCH_SYNC_RECORDS: _, /* @param Array.<string> categoryNames, @param {number} startAt (in seconds) */
   /**
    * webview -> browser
-   * browser must update its local values with the new sync records, performing
-   * conflict-resolution as necessary.
+   * browser should find objects with matching objectIds then call
+   * RESOLVE_SYNC_RECORDS to perform conflict resolution.
    */
   RECEIVE_SYNC_RECORDS: _, /* @param {string} categoryName, @param {Array.<Object>} records */
   /**
    * browser -> webview
+   * webview resolves sync records against matching browser objects and returns
+   * resolved syncRecords to apply against browser data.
+   * param looks like [[<syncRecord>, <browserObject=>], ...]
+   */
+  RESOLVE_SYNC_RECORDS: _, /* @param {Array.<Array.<Object>>} recordsAndExistingObjects */
+  /**
+   * webview -> browser
+   * browser must update its local values with the resolved sync records.
+   */
+  RESOLVED_SYNC_RECORDS: _, /* @param {Array.<Object>} records */
```

Comment:
given that this doesn't have a categoryName parameter, it seems the browser has to infer category from the record object anyway, so categoryName probably isn't needed for RECEIVE_SYNC_RECORDS either